ze/tracer_ze_helpers.include.c

```diff
@@ -274,9 +308,13 @@ static inline void _register_ze_event(
   if (o_h) {
     cl_data = (struct _ze_command_list_obj_data *)(o_h->obj_data);
     _ze_event->context = cl_data->context;
-  }
+    if (cl_data->flags & _ZE_IMMEDIATE)
+      _ze_event->flags |= _ZE_IMMEDIATE_CMD;
+  } else
+    THAPI_DBGLOG("Could not get command list associated to event: %p", event);
```

Comment:
Just by convension should will not always put `{}`arround macro?

Reply:
the macros are protected by `do { } while(0)` constructs for this specific reason.